base/compiler/ssair/slot2ssa.jl

```diff
@@ -420,10 +420,14 @@ function domsort_ssa!(ir, domtree)
     end
     result_stmts = Any[renumber_ssa!(result_stmts[i], inst_rename, true) for i in 1:length(result_stmts)]
     cfg = CFG(new_bbs, Int[first(bb.stmts) for bb in new_bbs[2:end]])
-    new_new_nodes = NewNode[(inst_rename[pos].id, reverse_affinity, typ,
-        renumber_ssa!(isa(node, PhiNode) ?
-        rename_phinode_edges(node, 0, result_order, bb_rename) : node,
-        inst_rename, true), lno) for (pos, reverse_affinity, typ, node, lno) in ir.new_nodes]
+    # (pos, reverse_affinity, typ, node, lno)
+    new_new_nodes = NewNode[
+        NewNode(entry,
+            pos = inst_rename[entry.pos].id,
+            node = renumber_ssa!(isa(entry.node, PhiNode) ?
+                rename_phinode_edges(entry.node, 0, result_order, bb_rename) : entry.node,
+                inst_rename, true)
+        ) for entry in ir.new_nodes]
```

Comment:
avoid creating a closure over `Box` variables here (see elsewhere for how to re-write this)